src/core/types/addons.js

```diff
@@ -99,6 +104,7 @@ export type ExternalAddonType = {|
   is_source_public: boolean,
   last_updated: Date,
   latest_unlisted_version: ?AddonVersionType,
+  locale_disambiguation?: string,
```

Comment:
When an `ExternalAddonType` is a language tool it will only have [a limited field set](https://addons-server.readthedocs.io/en/latest/topics/api/addons.html#language-tools). Because of this, it would be good to go through [`ExternalAddonType`](https://github.com/mozilla/addons-frontend/blob/74779218ba2a2a5522f0c1e5459ac5d93c041e62/src/core/types/addons.js#L86) and add question marks to all the fields that would be undefined in the case of a language tool. This will help us catch parts of our code that might throw an error for undefined field values.

However, if the Flow errors force you to write overly defensive code then we may need to take a different approach. Try it and see. The other approach is to use literal strings for `addon.type` but it will require opt'ing in more files to Flow and a bit of effort.

Reply:
Generally we will fetch that info on the details page, but I getcha. It's not trivial to make the change but I think it makes sense... I'll try for like twenty minutes or so.

Reply:
Most of the issues are around GUIDs though, which maybe do exist and we could get the API to send down.